src/components/ReactTags.js

```diff
@@ -339,16 +339,16 @@ class ReactTags extends Component {
     }
 
     // call method to add
-    this.props.handleAddition(tag);
-
-    // reset the state
-    this.setState({
-      query: '',
-      selectionMode: false,
-      selectedIndex: -1,
-    });
+    if (this.props.handleAddition(tag) !== false) {
```

Comment:
instead of returning `false` or `true` can you add a new prop `isTagValid` ? This prop when not passed from parent will return true as default prop and when present will be used to decide if `handleValidation` should be called ?